src/open_inwoner/accounts/signals.py

```diff
@@ -82,16 +84,49 @@ def _update_user_from_esuite(
 
 
 def _update_eherkenning_user_from_kvk_api(user: User):
-    kvk_client = KvKClient()
-
-    if user.vestiging:
-        vestiging = kvk_client.get_vestiging(vestiging=user.vestiging)
-    else:
-        vestiging = kvk_client.get_company_headquarters(kvk=user.kvk)
-
-    if company_name := vestiging.get("naam"):
-        user.company_name = company_name
-        user.save()
+    try:
+        kvk_client = KvKClient()
+        updated_fields = []
+
+        # Update company name. We want this to run on every login because it may
+        # change (if infrequently)
+        if vestiging := (
+            kvk_client.get_vestiging(vestiging=user.vestiging)
+            if user.vestiging
+            else kvk_client.get_company_headquarters(kvk=user.kvk)
+        ):
+            if (
+                company_name := vestiging.get("naam")
+            ) and user.company_name != company_name:
+                user.company_name = company_name
+                user.save(update_fields=["company_name"])
+                updated_fields.append("company_name")
+        else:
+            logger.error("Unable to connect to KvK API to retrieve vestiging")
+
+        # Optionally update RSIN. Unlike company name, RSIN should be immutable, so we
+        # only have to fetch it if it's not set.
+        if not user.rsin:
+            rsin = kvk_client.retrieve_rsin_with_kvk(user.kvk)
+
+            if rsin:
+                user.rsin = rsin
+                user.save(update_fields=["rsin"])
+                updated_fields.append("rsin")
+            else:
+                logger.error("Unable to connect to KvK API to retrieve rsin")
```

Comment:
It's not 
```suggestion
                logger.error("Unable to retrieve RSIN from KvK API")
```
It's not necessarily a connection error (perhaps also log the KVK number so that we know for which KVK the attempt failed)